### ISSUE TITLE
setup.py is broken if hg returns a date with timezone

### ISSUE THREAD

**Issue by blais** [beancount/beancount#304]
**[Original report](https://bitbucket.org/blais/beancount/issue/304) by Jason Chu (Bitbucket: [xentac](https://bitbucket.org/xentac), GitHub: [xentac](https://github.com/xentac)).**

----------------------------------------

After https://bitbucket.org/blais/beancount/commits/2b20d301782528147e0aba744da9a014323ef630 setup.py can fail if the vc_timestamp includes a timezone.

The stacktrace looks like this:

```
#!python

Traceback (most recent call last):
  File "setup.py", line 270, in <module>
    ('VC_TIMESTAMP', int(float(vc_timestamp))),
ValueError: could not convert string to float: '1527498118.0-32400' 
```

According to the hg book, {date} can return timezone as well (http://hgbook.red-bean.com/read/customizing-the-output-of-mercurial.html).

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

https://bitbucket.org/blais/beancount/commits/e93c8b2854a46acd2171d5ac2f7a89fdd3244144

**Comment by blais** [comment]
**Original comment by Jason Chu (Bitbucket: [xentac](https://bitbucket.org/xentac), GitHub: [xentac](https://github.com/xentac)).**

----------------------------------------

Looks like that change didn't fix it. From what I can tell |short is for {node} types.

For what it's worth, the build pipeline has been exhibiting the same problem (https://bitbucket.org/blais/beancount/addon/pipelines/home#!/).

Looking through the hg source, it looks like instead of 
```
#!

{date}
```
you can use 
```
#!

{date(date, "%s")}
```

Which also has the benefit of not being a float, so you can save yourself a cast.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

```
#!python

yuzu [hg|default]:~/p/beancount$ hg --version | head -n1
Mercurial Distributed SCM (version 4.6)
yuzu [hg|default]:~/p/beancount$ uname -a
Darwin yuzu 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
yuzu [hg|default]:~/p/beancount$ hg parent --template='{date}\n'
1527558969.0-32400
yuzu [hg|default]:~/p/beancount$ hg parent --template='{date|short}\n'
1527558969.0
yuzu [hg|default]:~/p/beancount$ hg parent --template='{date(date, "%s")}\n'
1527558969
```

I believe it should have fixed it (see above).
What versions are you using?

I'll go ahead and change it, and hope that your version is more robust across platforms.
Thanks for looking up the details.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #304, changed hg template to get rid of timezone offset on timestamp (thanks to Jason Chu).

**Comment by blais** [comment]
**Original comment by Jason Chu (Bitbucket: [xentac](https://bitbucket.org/xentac), GitHub: [xentac](https://github.com/xentac)).**

----------------------------------------

Just for the record, my output from those commands looks like this:

```
Mercurial Distributed SCM (version 4.3.1)                                                          
Linux 2c03d96315a6 4.14.44-coreos-r1 #1 SMP Thu May 31 06:04:02 UTC 2018 x86_64 Linux              
1527758505.0-32400
1527758505.0-32400                                                                                 
1527790905
```

It all runs within a docker instance, so the shell isn't interactive and the environment can be funnier than usual. This reminds me that I need to update to the new python base image...
